src/Configuration/ValueObject/ExportConfig.php

```diff
@@ -55,7 +58,8 @@ public static function fromArray(array $data): self
             $data['columns'],
             $data['outputTable'],
             $data['primaryKey'],
-            $data['retries']
+            $data['retries'],
+            !empty($data['ssl']) ? SSLConnectionConfig::fromArray($data) : null
```

Comment:
@ondrajodas , mam k tomu nejake otazky:

1. ssl konfiguracia nie je v `root.ssl` ale v  `root.db.ssl` nie? 
pozri: https://github.com/keboola/db-extractor-mysql#configuration-options

2. Neviem, ci v sucasnosti by malo byt `SSLConnectionConfig` sucastou `ExportConfig` ... 
z dovodu ze ked ides cez `configRow` tak tam tie data (`db.ssl`) mas k dispozicii ...  
ale ak ides cez `tables` tak ich tam nemas k dispoziicii ....  (v premennej `$data`)

    - pozri: https://github.com/keboola/db-extractor-common/blob/85b75a6e065a24b38e9c9bc1b841e4d4861af9cb/src/Application.php#L113-L123  
   - je to podobny problem ako som riesil ja pri value objects, ... ze bez nejakeho `adaptera` tam nemoze byt klasicky config.   
   - Teda podla mna by `SSLConnectionConfig` mala byt samostatna jednotka (teda tak ako je teraz, len nedavat to do `ExportConfig`):
    - potrebujes to dostat do `Extractor::createConnection` ... mozes to riesit v kazdom extractore osobitne ... alebo to das do `BaseExtractor::__construct` ... ak bude ssl konfiguracia, tak to ulozis do `$this->sslConfig`, ... ak nie tak tam bude `null` ... a potom to mozes v extractore hocikde pouzit (vacsinou asi v `testConnection`) .... (iba priklad, moze sa to riesit aj inak, napr. cez getter).

V buducnosti, ked spravime refactoring konfiguracie, ... nejaky adapter `tables[]` -> `configRow[]` ... a standardnu triedu `Config` ... tak sa z `ExportConfig` a `SSLConnectionConfig` mozu spravit interfacy, ktore bude ten standardny config `Config` implementovat.

Asi su aj nejake ine riesenia, ... skus si prejst `data-flow` konfiguracie v pripade `tables` a `config row` a porozmyslat ci to je bez problemov, ... imho sa to teraz neda pouzit ... ale mozno som nieco prehliadol.

Alternativne riesenie je upravit: https://github.com/keboola/db-extractor-common/blob/85b75a6e065a24b38e9c9bc1b841e4d4861af9cb/src/Application.php#L114 ...  
a pridat k `$table` aj `db` hodnoty (to by bol vlastne uz zarodok nejakeho toho adaptera) ... no cistejsie mi pride to prve riesenie co som spomenul ... kedze `SSLConnectionConfig` pouzijes iba pri vytvarani spojenia a inde su ti tie hodnoty zbytocne.

Napis mi teda prosim, ako to planujes riesit, aby som to vedel skontrolovat ;)
Alebo mozeme s tymto PR pockat, a pouzijes ho ako vyvojovu vetvu v dalsom PR 
a ked to bude vsetko spolu sediet, tak to mergneme a upravime `composer.json` .... je to na tebe.

Reply:
1/ no jasně, to máš pravdu

Reply:
2/ přemýšlel jsem udělat to tak jak popisuješ ve druhém případě (k tables hodnotám přidat i "db") příjde mi to tak lepší už jen z toho důvodu aby "ExportConfig" skutečně exportoval celý config a ne jen část (nebo minimálně jeho většinu)

Reply:
@ondrajodas ok, nemam nic proti ... oprav teda tu 1/, pozri este ostatne komentare, a hod ma potom opat na review ;)

Reply:
1/ přesunuto